core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java

```diff
@@ -63,7 +63,7 @@
                 bidirectionalActions = new BasicBidirectionalActions<>(fromInitialStatePart(), fromGoalStatePart());
                 return bidirectionalActions;
             }
-            if (isSolution(pathExistsBidirectional(back, exploredBack, exploredFront, originalProblem), false)) {
+            if (isSolution(pathExistsBidirectional(back, exploredBack, exploredFront, reverseProblem), false)) {
```

Comment:
Though ```originalProblem``` would have worked (just like it already is) because the same graph is used for both the problems (original and reverse) with only the initial state differing, ```reverseProblem``` makes it semantically clearer and so works good for the purpose.